src/Pact/Analyze/Eval/Term.hs

```diff
@@ -288,28 +304,53 @@ evalTerm :: (Show a, SymWord a) => Term a -> Analyze (S a)
 evalTerm = \case
   CoreTerm a -> evalCore a
 
-  IfThenElse cond then' else' -> do
+  IfThenElse cond (thenPath, then') (elsePath, else') -> do
+    reachable <- use purelyReachable
     testPasses <- evalTerm cond
-    iteS testPasses (evalTerm then') (evalTerm else')
+    tagSubpathStart (reachable &&& testPasses) thenPath
+    tagSubpathStart (reachable &&& bnot testPasses) elsePath
+    iteS testPasses
+      (evalTerm then')
+      (evalTerm else')
 
   -- TODO: check that the body of enforce is pure
-  Enforce cond -> do
+  Enforce mTid cond -> do
     cond' <- evalTerm cond
+    maybe (pure ()) (\tid -> tagAssert tid cond') mTid
     succeeds %= (&&& cond')
     pure true
 
-  EnforceOne conds -> do
+  EnforceOne (Left tid) -> do
+    tagAssert tid false -- in this case (of an empty list), we always fail.
+    succeeds .= false
+    pure true           -- <- this value doesn't matter.
+
+  -- TODO: check that each cond is pure. checking that @Enforce@ terms are pure
+  -- does *NOT* suffice; we can have arbitrary expressions in an @enforce-one@
+  -- list.
+  EnforceOne (Right conds) -> do
     initSucceeds <- use succeeds
 
-    successRecord <- for conds $ \cond -> do
-      succeeds .= true
-      _ <- evalTerm cond
-      use succeeds
+    (result, anySucceeded) <- foldlM
+      (\(prevResult, earlierSuccess) ((failTag, passTag), cond) -> do
+        succeeds .= true
+        res <- evalTerm cond
+        currentSucceeded <- use succeeds
 
-    let anySucceeded = bOr successRecord
-    succeeds .= (initSucceeds &&& anySucceeded)
+        let reachable = bnot earlierSuccess
+        tagSubpathStart (reachable &&& currentSucceeded)      passTag
+        tagSubpathStart (reachable &&& bnot currentSucceeded) failTag
 
-    pure true
+        pure $ iteS earlierSuccess
+          (prevResult, true)
+          (iteS currentSucceeded
```

Comment:
i think this works? `(res, iteS currentSucceeded)`

Reply:
Yeah good call -- changing it to `(res, currentSucceeded)`